### PR TITLE
Test for duplicate plugin configuration parameters in completion

### PR DESCRIPTION
**Pull request by AObuchow**
Part of #110

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

### CODE REVIEW

**Comment by AObuchow** [comment]
Test is intended to fail currently.

File: lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenCompletionParticipant.java [hunk 1]
```diff
@@ -265,13 +265,6 @@ public void onXMLContent(ICompletionRequest request, ICompletionResponse respons
 		case "goal":
 			collectGoals(request).forEach(response::addCompletionItem);
 			break;
-		case "configuration":
```
Comment (AObuchow) [inline lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenCompletionParticipant.java:268]: @mickaelistria it seems this wasn't necessary, tests are passing without it and things seemed to work when user testing. Any worries about removing this?

**Comment by mickaelistria** [comment]
So do I get it right that when completing

```xml
  <configuration>
     |(completeHere)
  </configuration>
```
LemMinX invokes both onXMLContent and onTagOpen to gather the results?
If so, it's a good thing LemMinX does that, and feel free to remove this piece of code if it causes bugs. Maybe we can also identify other cases that belong to this pattern then, and move them from onXMLCotent to onTagOpen, it seems like anything that inserts a tag could be a good candidate.

**Comment by AObuchow** [comment]
@mickaelistria that seems correct to me, based on my understanding. @angelozerr could you maybe confirm?

**Pull request merged**